agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -433,6 +434,60 @@ impl DeriveSchema for Stage {
             }
         }
 
+        fn rank_fusion_derive_schema(
+            rank_fusion: &RankFusion,
+            state: &mut ResultSetState,
+        ) -> Result<Schema> {
+            // 1. Derive the schema for each pipeline and union them together
+            let unioned_schema_pipelines: Schema = rank_fusion
```

Comment:
could we make `unioned_schema_pipelines` mutable and then just always return it at the end (e.g. in the if case, just do `unioned_schema_pipelines = unioned_schema_pipelines.union(score_details_metadata_schema)`)?